src/coreclr/interpreter/ee_il_dll.cpp

```diff
@@ -0,0 +1,142 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+#include <stddef.h>
+#include "corjit.h"
+
+#include "interpreter.h"
+#include "ee_il_dll.hpp"
+
+#include <string.h>
+#include <stdio.h>
+
+#ifdef _MSC_VER
+#define DLLEXPORT
+#else
+#define DLLEXPORT __attribute__ ((visibility ("default")))
+#endif // _MSC_VER
+
+#include <vector>
+
+// FIXME We will probably end up not needing this table.
```

Comment:
If this .dll should act like a special JIT, it should not have any global state. All global state should be on the VM side.

Reply:
Wasn't really keen on further refactoring for this table which has a dubious use case anyway. I added another commit where I embed the `InterpMethod` pointer at the start of the code, making easier to obtain this information directly from compiled code, without further indirection through `MethodDesc`. In the future it might be best for it to be stored in some kind of header for interpreter code, probably once we have a different code manager for interpreter code.